### PR TITLE
[werft/observability] Completely recycle CRDs instead of incremental applies

### LINKED ISSUE #6455: [preview-envs/observability] Prometheus CRD fails to install
Monitoring satellite fails 100% of the time right now because it fails to apply the Prometheus Custom Resource Definition into the cluster, with the following error:
```
The CustomResourceDefinition "prometheuses.monitoring.coreos.com" is invalid: metadata.annotations: Too long: must have at most 262144 bytes
```

### PR DESCRIPTION
**Pull request by ArthurSens**
## Description

After https://github.com/prometheus-operator/prometheus-operator/pull/4326, Prometheus CRD has become too big to fit into the "kubectl.kubernetes.io/last-applied-configuration" annotation, therefore we cannot use `kubectl apply` anymore.

## Related Issue(s)

Fixes #6455 

## How to test

Start a werft job with `with-observability` annotation. CRDs should be replaced without errors and the observability stack should complete.

## Release Notes

```release-note
NONE
```

### CODE REVIEW

File: .werft/observability/monitoring-satellite.ts
Comment (mads-hartmann) [inline .werft/observability/monitoring-satellite.ts:198]: It is preferred to use `const` whenever a variable never needs to be assigned :)

File: .werft/observability/monitoring-satellite.ts
Comment (mads-hartmann) [inline .werft/observability/monitoring-satellite.ts:205]: This exec is different from the one in the `if` branch. Do they install different things? If not then it might be easier to simplify it so that it goes like this 

**If CRDs have to be installed separately from the rest**

```ts
if (crdExists) {
  // Delete CRDs
}
// Install CRDs
// Install non-CRDs
```

**If CRDs don't have to be installed separately from the rest**

```ts
if (crdExists) {
  // Delete CRDs
}
// Install prometheus operator
```

That way you don't need to install CRDs in both the `if` and `else` branch :)

File: .werft/observability/monitoring-satellite.ts
Comment (mads-hartmann) [inline .werft/observability/monitoring-satellite.ts:209]: If CRDs don't have to be installed separately then this can be simplified, see comment above.

**Review by mads-hartmann — CHANGES_REQUESTED** [review]
I think it might be possible to simplify the script. If I have misunderstood something let me know and I'll approve ☺️

File: .werft/observability/monitoring-satellite.ts
Comment (ArthurSens) [reply .werft/observability/monitoring-satellite.ts:205]: CRDs do need to be created first! If Prometheus-operator is created without the CRDs installed then it remains in crashloop until someone manually restarts it

Comment (ArthurSens) [reply .werft/observability/monitoring-satellite.ts]: And thanks for the reminder of excluding unnecessary `else` 🙂

Comment (mads-hartmann) [reply .werft/observability/monitoring-satellite.ts]: @ArthurSens Thanks for clarifying why we need to install CRDS first 🤗

**Comment by mads-hartmann** [comment]
@ArthurSens Just re-request a review whenever this is ready and I'll take another look :)

**Review by mads-hartmann — APPROVED** [review]
Awesome 🚀

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6458#pullrequestreview-796255993" title="Approved">mads-hartmann</a>*

Associated issue: *#6455*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[.werft/observability/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/.werft/observability/OWNERS)~~ [mads-hartmann]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ArthurSens** [comment]
@mads-hartmann, only lgtm left :)

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 81d75980a30bc24ca6fd126634b1243af0d53638</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6458#issuecomment-959348564" title="LGTM">mads-hartmann</a>*

Associated issue: *#6455*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[.werft/observability/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/.werft/observability/OWNERS)~~ [mads-hartmann]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
